test/e2e/app-dir/next-form/default/pages/pages-dir/forms/with-function/action-client/index.tsx

```diff
@@ -0,0 +1,33 @@
+'use client'
+import * as React from 'react'
+import { useActionState, useState } from 'react'
+import Form from 'next/form'
+import { useRouter } from 'next/router'
+
+export default function Page() {
+  const destination = '/pages-dir/redirected-from-action'
+  const router = useRouter()
+  const [, dispatch] = useActionState(() => {
+    const to = destination + '?' + new URLSearchParams({ query })
+
+    // TODO(form): this doesn't work unless wrapped in a startTransition...
+    // why is this necessary here?
+    // without it, the URL updates, but we stay on the old page...
+    return React.startTransition(() => {
+      router.push(to)
+    })
+  }, undefined)
```

Comment:
not quite sure why this is needed

Reply:
no, typescript is irrelevant. what i mean is that unless we wrap that navigation in `startTransition`, it won't happen at all

Reply:
i've removed all the transition-related stuff from here. the code here wasn't actually doing anything. leaving all transition-related work for the follow up here https://github.com/vercel/next.js/pull/70681